services/helper.py

```diff
@@ -10,11 +10,23 @@ def __init__(self, repository: HelperRepository, logger: logging.Logger):
         self.logger = logger
         self.repository = repository
 
-    def change_points(self, thanked, thanker, bot, p):
+    async def change_points(self, thanked, thanker, bot, p):
         if thanked not in {thanker, bot}:
             self.repository.add_points_to_user(thanked.id, p)
-            self.logger.debug(f'User {thanked.display_name} {"lost" if p < 0 else "got"} {p} point{"" if abs(p) == 1 else "s"} from '
+            role_id = self.repository.role_id_user_should_have(thanked.id)
+            await self.change_roles(thanked, role_id)
+            self.logger.debug(f'User {thanked.display_name} {"lost" if p < 0 else "got"} {abs(p)} point{"" if abs(p) == 1 else "s"} from '
                               f'{thanker.display_name}')
         else:
             self.logger.debug(f'User {thanker.display_name} cannot thank '
                               f'{thanked.display_name}')
+
+    async def change_roles(self, user, role_id):
```

Comment:
Trochę nie rozumiem jak działa ta zmiana roli w tej funkcji, mógłbyś to po krótce objaśnić?

Reply:
Funkcja najpierw usuwa użytkownikowi wszystkie zbędne role związane z pomocą i później daje poprawną rolę, chyba że użytkownik już ma tą rolę to wtedy nic się nie zmienia